### DISCUSSION TITLE
IP Prefix Contact Assignment

### DISCUSSION THREAD

**Discussion by LukasIgel** [netbox-community/netbox#11930] [Help Wanted!]
Dear community,

is there any way to use Contact Assigment with IP prefix?

I have documented employees and the hierachical structure of my corporation, aka the departments, by Contacts and Contact Groups. Now I want to assign a functional "group" of contacts to an IP Prefix. This "group" would consist of a select group of employees from multiple departments.

I had hoped that I can use Contact Roles to implement these functional "groups". But the only way (I know of) to use Contact Roles  is by using Contact Assignment, which is not available for IP Prefix.

I cannot use the models Tenant and Tenant Groups.

I have already tried using Custom Fields, but that way I can only assign already existing objects of Contact Assigment to IP Prefixes.

**Comment by Kludge-As-A-Service** [comment]
I am not sure I understand your problem with custom fields.
If it's a matter of "too many clicks in too many places to accomplish the task", you can maybe smoothen it up with a custom script that would show the user a form asking for the relevant information and would create the various objects automatically.

**Reply by LukasIgel** [reply]
Thanks for the reply. When I create a custom field on Prefix of type Contact Assignment, I can't actually assign Contacts with a specific Contact Role. I can only assign existing Contact Assignments made on different object types.

This is a picture of howContact Assignments work for Sites (I want the same for Prefix):
![image](https://user-images.githubusercontent.com/95239141/224287795-5bfa05e2-67cd-40f4-95fd-4bfe545b2458.png)

When you click on "Add a contact", you can create a new Assignment object by selecting a Contact and its Role:
![image](https://user-images.githubusercontent.com/95239141/224288051-29d6f16e-303c-4d0c-ab7a-071304935d2b.png)

Here's how it looks when I create a custom field (I can only search for existing Contact Assignment objects):
![image](https://user-images.githubusercontent.com/95239141/224285670-d8e3295d-c4c3-499f-85c6-9e7d304a3a64.png)

**Reply by Kludge-As-A-Service** [reply]
Ah I see. Yeah as I understand it (though I am not an expert) it seems like it's not possible to create contact assignments this way. 
Wouldn't it be sufficient to have your custom field reference the contact group directly instead?

**Reply by LukasIgel** [reply]
That may be a solution. I originally wanted to use Contact Groups to display the hierarchical structure of my corporation and its departments, but the functional grouping is more important.

Another idea: Tenant has a native relation to Contact Assignment. Maybe I create a tenant for each prefix and then do the contact assignment for the prefix through the tenant. I just don't know if that's the correct way to use these data models.

**Reply by Kludge-As-A-Service** [reply]
Another option is to do it the other way around and create a custom field for your contacts, of the "Multiple objects" variant, referencing prefix objects. So that you can configure which person manages which prefixes. One limitation I see with this approach is that UI-wise, it seems like you can't simply add/remove prefixes managed by several users like you can do with tags. But a custom script could handle that gracefully. 

Or you could of course go with the "only tags" approach, same as previous option but instead of a custom field it's a tag referencing prefix name. This may require an external way to ensure data consistency as we're not referencing actual objects anymore, but UI-wise it's easy to work with.
